### PR TITLE
Remove duplicate app.config entries Fixes #7451

### LINKED ISSUE #7451: Dupicate entries for Microsoft.Build.Conversion.Core in app.config files
app.config and app.amd64.config contain duplicate antries for Microsoft.Build.Conversion.Core.

We should delete the duplicate entry and hope nothing breaks (it's been there for a while) 🤞

### PR DESCRIPTION
**Pull request by Forgind**
Fixes #7451

### Context
There are duplicate entries for Microsoft.Build.Conversion.Core in our app.config and app.amd64.config files. Remove those.

### Changes Made
Removed unnecessary duplicate entries.

### Testing
None

### CODE REVIEW

File: src/MSBuild/app.amd64.config [hunk 1]
```diff
@@ -45,11 +45,6 @@
           <bindingRedirect oldVersion="0.0.0.0-99.9.9.9" newVersion="15.1.0.0" />
           <codeBase version="15.1.0.0" href="..\Microsoft.Build.Engine.dll"/>
         </dependentAssembly>
-        <dependentAssembly>
-          <assemblyIdentity name="Microsoft.Build.Conversion.Core" culture="neutral" publicKeyToken="b03f5f7f11d50a3a" />
-          <bindingRedirect oldVersion="0.0.0.0-99.9.9.9" newVersion="15.1.0.0" />
-          <codeBase version="15.1.0.0" href="..\Microsoft.Build.Conversion.Core.dll"/>
```
Comment (BenVillalobos) [inline src/MSBuild/app.amd64.config:51]: We'll need the codebase entry in the leftover item for `Microsoft.Build.Conversion.Core`

**Review by BenVillalobos — APPROVED** [review]
Approved, pending the codebase entry

File: src/MSBuild/app.amd64.config (hunk 1 above)
Comment (Forgind) [reply src/MSBuild/app.amd64.config]: You'd actually already added that in #7472, and I avoided a merge conflict here. I switched which I deleted (and accepted the merge conflict) to make that clearer.

**Comment by rainersigwald** [comment]
@arahaan auto-generated from what?

**Review by rainersigwald — APPROVED** [review]

**Comment by Forgind** [comment]
If you're thinking of the GenerateBindingRedirects task, that's part of MSBuild (and relies on other parts of MSBuild like RAR), so even if we had a clear set of what binding redirects to generate, we'd have to execute MSBuild code in order to find/find the correct version of the MSBuild code we'd need to execute. Can't fix that cycle.

**Pull request merged**